vendor/api/v1/stripe/views.py

```diff
@@ -0,0 +1,193 @@
+            credentials = StripeIntegration(site)
+
+            if credentials.instance and credentials.instance.private_key:
+                self.event = stripe.Event.construct_from(json.loads(self.request.body), credentials.instance.private_key)
+            elif settings.STRIPE_PUBLIC_KEY:
+                self.event = stripe.Event.construct_from(json.loads(self.request.body), settings.STRIPE_PUBLIC_KEY)
+            else:
+                return False
+            
+        except ValueError as exce:
+            return False
+        except Exception as exce:
+            return False
+
+        return True
+    
+    def is_incoming_event_correct(self, event, desired_event):
+        if event.type != desired_event:
+            return False
+
+        # This check is recuired to make sure that the event is related to a subscription.
+        if self.event.data.object.billing_reason != 'subscription_cycle':
+            return False
+
+        return True
+
+
+class StripeSubscriptionInvoicePaid(StripeBaseAPI):
+
+    def post(self, request, *args, **kwargs):
+        site = get_site_from_request(self.request)
+
+        if not self.is_valid_post(site):
+            return HttpResponse(status=400)
+
+        if not self.is_incoming_event_correct(self.event, StripeEvents.INVOICE_PAID):
+            return HttpResponse(status=400)
+
+        stripe_invoice = self.event.data.object
+        paid_date = timezone.datetime.fromtimestamp(stripe_invoice.status_transitions.paid_at)
+        amount_paid = convert_integer_to_float(stripe_invoice.total)
+
+        # TODO nice to move this try/except blocks inside a generic function in the StripeBaseAPI class
+        try:
+            customer_profile = CustomerProfile.objects.get(site=site, user__email__iequals=stripe_invoice.customer_email)
+        except ObjectDoesNotExist:
+            logger.error(f"StripeSubscriptionInvoicePaid error: email: {stripe_invoice.customer_email} does not exist")
+            return HttpResponse(status=400)
+
+        if 'stripe_id' not in customer_profile.meta:
+            customer_profile.meta['stripe_id'] = stripe_invoice.customer
+            customer_profile.save()
+
+        try:
+            subscription = Subscription.objects.get(meta__stripe_id=stripe_invoice.subscription, profile=customer_profile)
+        except ObjectDoesNotExist:
+            logger.error(f"StripeSubscriptionInvoicePaid customer: {customer_profile} does not have a subscription with stripe_id: {stripe_invoice.subscription}")
+            return HttpResponse(status=400)
+
+        offer = subscription.get_offer()
+
+        if not offer:
+            logger.error(f"StripeSubscriptionInvoicePaid {subscription} has no offer attached")
+            return HttpResponse(status=400)
+
+        invoice = Invoice.objects.create(
+            profile=customer_profile,
+            site=site,
+            ordered_date=paid_date,
+            total=float(f"{amount_paid[:-2]}.{amount_paid[-2:]}"),
```

Comment:
should this just use amount_paid? it looks like you already did the float conversion above

Reply:
dam it I thought I changed that, change pushed sorry and thank you